### PR TITLE
Fixes #780 : Added CSV and Text area as separate components

### LINKED ISSUE #780: Refactoring the file upload form component in frontend
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behaviour:**

CSV is not uploaded and unable to send to the backend.

**Expected behaviour:**

File should get uploaded and sent to backend server.

__Working__ on the same.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #780

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Added CSV upload and text area component and corresponding actions.
-

### CODE REVIEW

**Comment by yashLadha** [comment]
Ping @harshit98 @djmgit @gabru-md @ParthS007 @vaibhavsingh97 . Please review it

File: frontend/app/templates/components/data-area.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    <div class="field">
+        <label>Upload CSV</label>
+        <input name="csv" onchange={{action "encodeCSV"}} type="file">
+    </div>
+    <div class="field">
+        <label>Enter details</label>
+        {{textarea name="data" rows="3" value=formData}}
+    </div>
+    <button {{action "uploadForm"}}>Submit Data</button>
+</form>
```
Comment (vaibhavsingh97) [inline frontend/app/templates/components/data-area.hbs]: Add blank line

File: frontend/app/templates/components/data-component.hbs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<form class="ui form">
+    {{ yield }}
+</form>
```
Comment (vaibhavsingh97) [inline frontend/app/templates/components/data-component.hbs]: same

File: frontend/app/templates/components/data-component/csv-component.hbs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<div class="field">
+    <label>Upload CSV</label>
+    <input name="csv" onchange={{action "encodeCSV"}} type="file">
+</div>
```
Comment (vaibhavsingh97) [inline frontend/app/templates/components/data-component/csv-component.hbs]: same

File: frontend/app/templates/components/data-component/text-component.hbs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<div class="field">
+    <label>Enter details</label>
+    {{textarea name="data" rows="3" value=formData}}
+</div>
```
Comment (vaibhavsingh97) [inline frontend/app/templates/components/data-component/text-component.hbs]: same

**Comment by harshit98** [comment]
Hi @yashLadha,

 Kindly do the changes in the PR suggested by @vaibhavsingh97. The PR looks good to me. I'll merge it after you're done with the changes.

Cheers!

**Comment by yashLadha** [comment]
Done with the changes @harshit98

**Pull request merged**